### PR TITLE
[3.2.x] Fixed #33062 -- Make MultiPartParser ignore null bytes in file name

### PR DESCRIPTION
**Pull request by asedmammad**

### CODE REVIEW

File: tests/file_uploads/tests.py [hunk 1]
```diff
@@ -38,6 +38,7 @@
     '../..\\hax0rd.txt',        # Relative path, mixed.
     '..&#x2F;hax0rd.txt',       # HTML entities.
     '..&sol;hax0rd.txt',        # HTML entities.
+    '\x00hax0rd\x00.txt\x00',   # Null bytes.
```
Comment (kezabelle) [inline tests/file_uploads/tests.py:41]: Is there a reason this is in the `CANDIDATE_TRAVERSAL_FILE_NAMES` rather than the `CANDIDATE_INVALID_FILE_NAMES`?

**Review by felixxm — COMMENTED** [review]
@asedmammad Thanks for the patch :+1: Please re-target your branch to the `main`.

**Comment by apollo13** [comment]
Wouldn't it make sense to strip out all unprintable chars (or maybe control chars)? I do not want to limit unnecessarily but I can't see how useful a carriage return would be in a filename etc…

**Comment by kezabelle** [comment]
> Wouldn't it make sense to strip out all unprintable chars (or maybe control chars)? I do not want to limit unnecessarily but I can't see how useful a carriage return would be in a filename etc…

Though it expands on the ticket, I think there's a decent case for doing something like that. Looking at [Wikipedia's notes on filesystem limitations](https://en.wikipedia.org/wiki/Filename#Comparison_of_filename_limitations) it would seem that Windows (via FAT and NTFS) have the lion's share of reserved characters; most unixy ones are just `no nulls` (eg: [Ext4](https://en.wikipedia.org/wiki/Ext4)). Whilst I don't think Windows is a popular deployment target, it's probably a reasonable _development_ target, and having files fixed based on a lowest-common-denominator of valid characters would remove a -- probably baffling -- "works on my machine" scenario.

Counter to that argument is presumably the fact it's a niche concern, given its taken until _now_ for nulls to even make the shortlist.

**Comment by felixxm** [comment]
@asedmammad Do you have time to keep working on this?

**Comment by felixxm** [comment]
Closing due to inactivity.

**Pull request closed without merging**